### PR TITLE
`executor_kernel_arg` cleanup

### PR DESCRIPTION
**Pull request by zasdfgbnm**
This PR remove `ArgAbstract` and all its subclasses. `KernelArgumentHolder` now directly holds arguments as `PolymorphicValue`s. With this cleanup, it is now trivial to support various dtypes as fusion inputs, so I added a new test `NVFuserTest.AllInputDtypes` to test this.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

**Review by jacobhinkle — APPROVED** [review]
Nice simplification. I had one small comment.

File: csrc/executor.cpp [hunk 1]
```diff
@@ -884,11 +884,7 @@ std::vector<at::Tensor> allocOutputs(
     // for kernel execution, so would need to push them to args
     if (alias_it != output_to_input_aliases.end()) {
       auto aliased_input_index = alias_it->second;
-      auto tensor_arg_abstract = dynamic_cast<const TensorArgAbstract*>(
-          inputs.at(aliased_input_index));
-      TORCH_INTERNAL_ASSERT(
-          tensor_arg_abstract, "alias io only supports tensor");
-      outputs.emplace_back(tensor_arg_abstract->getTensor());
+      outputs.emplace_back(*inputs[aliased_input_index]);
```
[the remark this answers is not in the document]
Comment (zasdfgbnm) [reply csrc/executor.cpp:890]: I agree. Added

**Comment by zasdfgbnm** [comment]
!build

**Pull request merged**